SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,192 @@
+      final BaragonServiceState baragonServiceState = singularityCheckingUpstreamsUpdate.getBaragonServiceState().get();
+      return baragonServiceState.getUpstreams();
+    }
+    LOG.info("Baragon service state for service {} is absent.", singularityCheckingUpstreamsUpdate.getSingularityRequestId());
+    return Collections.emptyList();
+  }
+
+  private Collection<UpstreamInfo> getLoadBalancerUpstreamsForService(String singularityRequestId) {
+    LOG.info("Sending request to get load balancer upstreams for service {}.", singularityRequestId);
+    try {
+      final SingularityCheckingUpstreamsUpdate checkUpstreamsState = lbClient.getLoadBalancerServiceStateForRequest(singularityRequestId);
+      LOG.info("Succeeded getting load balancer upstreams for singularity request {}. State is {}.", singularityRequestId, checkUpstreamsState.toString());
+      return getLoadBalancerUpstreamsForServiceHelper(checkUpstreamsState);
+    } catch (Exception e) {
+      LOG.error("Failed getting load balancer upstreams for singularity request {}. ", singularityRequestId, e);
+    }
+    return Collections.emptyList();
+  }
+
+  private Optional<SingularityLoadBalancerUpdate> syncUpstreamsForServiceHelper(SingularityRequest singularityRequest, SingularityDeploy deploy, Optional<String> loadBalancerUpstreamGroup) {
+    LOG.info("Sending load balancer request to sync upstreams for service {}.", singularityRequest.getId());
+    final LoadBalancerRequestId loadBalancerRequestId = new LoadBalancerRequestId(String.format("%s-%s-%s", singularityRequest.getId(), deploy.getId(), System.currentTimeMillis()), LoadBalancerRequestType.REMOVE, Optional.absent());
+    final long start = System.currentTimeMillis();
+    try {
+      Collection<UpstreamInfo> upstreamsInLoadBalancerForRequest = getLoadBalancerUpstreamsForService(singularityRequest.getId());
+      LOG.info("Upstreams in load balancer for service {} are {}.", singularityRequest.getId(), upstreamsInLoadBalancerForRequest);
+      Collection<UpstreamInfo> upstreamsInSingularityForRequest = getUpstreamsFromActiveTasksForRequest(singularityRequest.getId(), loadBalancerUpstreamGroup);
+      LOG.info("Upstreams in singularity for service {} are {}.", singularityRequest.getId(), upstreamsInSingularityForRequest);
+      final List<UpstreamInfo> extraUpstreams = getExtraUpstreamsInLoadBalancer(upstreamsInLoadBalancerForRequest, upstreamsInSingularityForRequest);
+      if (extraUpstreams.size() > 0){
+        LOG.info("Syncing upstreams for service {}. Making and sending load balancer request {} to remove {} extra upstreams. The upstreams removed are: {}.", singularityRequest.getId(), loadBalancerRequestId, extraUpstreams.size(), extraUpstreams);
+        return Optional.of(lbClient.makeAndSendLoadBalancerRequest(loadBalancerRequestId, Collections.emptyList(), extraUpstreams, deploy, singularityRequest));
```

Comment:
Should this have a separate catch around it? I feel like there are two cases we are lumping together here:
1) We catch an exception getting the upstreams (i.e. before we `makeAndSendLoadBalancerRequest`). Could log it and return Optional.absent() there since there is no request to poll for results
2) We catch an exception from `makeAndSendLoadBalancerRequest`, which can fall to the `BaragonRequestState.UNKNOWN` case you have in the current catch, since we don't know if the lb update went through or not